server/routers/prediction.py

```diff
@@ -42,111 +67,130 @@ async def get_all_prediction_models():
     all_models = get_models_db()
     return {'models': all_models}
 
+
 @model_router.get("/tags", dependencies=[Depends(current_user_investigator)])
-async def get_available_prediction_models():
+async def get_prediction_model_tags():
     """
     Returns list of tags for each available model
     """
     return {"tags": settings.models_tags}
 
+@model_router.get("/types", dependencies=[Depends(current_user_investigator)])
+async def get_prediction_model_types():
+    """
+    Returns list of types for each available model
+    """
+    return {"types": settings.model_types}
+
 
 @model_router.post("/predict")
-def create_new_prediction_on_image(images: List[UploadFile] = File(...),
-                                   models: List[str] = (),
-                                   current_user: User = Depends(current_user_investigator)):
+def create_new_prediction(model_type: str,
```

Comment:
So, this is the best way I could think of to add in type-checks. i.e. if someone is uploading things via the `Import Video` page vs the `Import Image` page,  this should be clear to the back-end. We also need to save the type of the file with the `UniversalMLPredictionObject` so the best way to get that is from the front-end that already makes sure the files are video, image, etc. Lmk if that makes sense.

Reply:
Ahhh ok that makes sense. Thanks for clarifying. I would recommend including this signature as an object in dependency.py as it is starting to get cluttered. Additionally, it will also keep everything in the request body if you make an object instead of having the loose string fields like model_type be a param.

For example:
```python
class PredictionRequest(BaseModel):
    objects: List[UploadFile] = File(...)
    models: List[str] = ()
    model_type: str = ''
```

and then the method in prediction.py becomes:
```python
def create_new_prediction(p: PredictionRequest, current_user: User = Depends(current_user_investigator)):
.......
```

Reply:
Ahh, okay thanks! Was wondering how to put that param in the request body. Will do that.